yo/app/scripts/controllers/model.js

```diff
@@ -436,6 +450,59 @@ angular.module('dmpApp')
             }
         }
 
+        function openSelectRecords(project) {
+
+            var recordsSelected = true;
+
+            if (!project._$selectedRecords) {
+                project._$selectedRecords = [];
+                recordsSelected = false;
+            }
+
+            var modalInstance = $modal.open({
+                templateUrl: 'views/directives/select-records.html',
+                controller: 'SelectRecordsCtrl',
+                windowClass: 'wide',
+                resolve: {
+                    project: function() {
+                        return project;
+                    },
+                    recordsSelected: function() {
+                        return recordsSelected;
+                    }
+                }
+            });
+
+            modalInstance.result.then(function(reason) {
+
+                if(reason && (reason.removeFilter || reason.resetRecordSelection)) {
+
+                    project._$selectedRecords = null;
+                    project.selected_records = null;
```

Comment:
I agree. _$selectedRecords is used with [], null and some values. Better check for length and always use an array.